File: app/src/main/java/com/softartdev/notecrypt/ui/signin/SignInPresenter.java
package com.softartdev.notecrypt.ui.signin;

import com.softartdev.notecrypt.App;
import com.softartdev.notecrypt.db.DbStore;

import javax.inject.Inject;

public class SignInPresenter {
    private SignInView mView;

    @Inject
    DbStore dbStore;

    SignInPresenter(SignInView signInView) {
        App.getAppComponent().inject(this);
        mView = signInView;
    }

    void signIn(String pass) {
        mView.hideError();
        if (pass.length() == 0) {
            mView.showEmptyPassError();
        } else if (checkPass(pass)) {
            mView.navMain();
        } else {
            mView.showIncorrectPassError();
        }
    }

    private boolean checkPass(String pass) {
        return dbStore.checkPass(pass);
    }
}

File: app/src/main/java/com/softartdev/notecrypt/ui/signin/SignInPresenter.kt
package com.softartdev.notecrypt.ui.signin

import com.softartdev.notecrypt.data.DataManager
import com.softartdev.notecrypt.di.ConfigPersistent
import com.softartdev.notecrypt.ui.base.BasePresenter
import javax.inject.Inject

@ConfigPersistent
class SignInPresenter @Inject
constructor(private val dataManager: DataManager) : BasePresenter<SignInView>() {

    override fun attachView(mvpView: SignInView) {
        super.attachView(mvpView)
    }

    fun signIn(pass: String) {
        mvpView!!.hideError()
        if (pass.isEmpty()) {
            mvpView!!.showEmptyPassError()
        } else if (checkPass(pass)) {
            mvpView!!.navMain()
        } else {
            mvpView!!.showIncorrectPassError()
        }
    }

    private fun checkPass(pass: String): Boolean {
        return dataManager.checkPass(pass)
    }
}
